corehq/apps/domain/models.py

```diff
@@ -880,6 +880,22 @@ def delete(self):
         super(Domain, self).delete()
         Domain.get_by_name.clear(Domain, self.name)  # clear the domain cache
 
+    def _delete_messaging_data(self):
+        from corehq.apps.sms.models import (SQLMobileBackendMapping,
+            MobileBackendInvitation, SQLMobileBackend, SMS, MessagingEvent,
+            MessagingSubEvent, SelfRegistrationInvitation)
+        # Ok to keep these as bulk deletes since all domain-related couch docs
+        # are deleted via bulk delete in self._pre_delete() (so we don't need the
+        # couch-sql sync to run on each object we're deleting)
+        SMS.objects.filter(domain=self.name).delete()
+        MessagingSubEvent.objects.filter(parent__domain=self.name).delete()
+        MessagingEvent.objects.filter(domain=self.name).delete()
+        SelfRegistrationInvitation.objects.filter(domain=self.name).delete()
+        SQLMobileBackendMapping.objects.filter(is_global=False, domain=self.name).delete()
+        MobileBackendInvitation.objects.filter(domain=self.name).delete()
+        SQLMobileBackend.objects.filter(is_global=False, domain=self.name).delete()
```

Comment:
is that for performance or organization? it seems like code-wise the `ModelDeletion`s are doing the same thing